### DISCUSSION TITLE
How do I use a newer IRremoteESP8266?

### DISCUSSION THREAD

**Discussion by voltagex** [arendst/Tasmota#12224] [Q&A] [answered]
How do I use a newer IRremoteESP8266 library?

I'm pretty sure v2.7.18 now supports my air conditioner, but Tasmota has a copy of 2.7.16 at https://github.com/arendst/Tasmota/tree/development/lib/lib_basic/IRremoteESP8266 (also, why not use submodules?)

**Comment by voltagex** [comment, marked as answer]
Yes, you can just drop in the newer library.

**Comment by Jason2866** [comment]
The changes from v.2.7.16 to v.2.7.18 are minimal. Tasmota do not use submodules or any extern libs, since many libs are modified. It is easier to maintain to have all libs "on board".
Doing this way no surprises can happen.

**Comment by s-hadinger** [comment]
Yes just drop in the new lib. I will do the update on Tasmota repo.

I missed those updates

**Comment by s-hadinger** [comment]
Also the reason of not using submodules is to be able to customize compile options with library.json. I have plans to remove Tasmota-ircustom with the help of some specific options

**Reply by voltagex** [reply]
wait, what impact will removing tasmota-ircustom have? My devices are really low on storage so I have to be careful what's compiled in.

**Reply by s-hadinger** [reply]
No impact on storage. It's just to be able to compile with full IR without requiring the Tasmota-ircustom artifact. You will need only standard Tasmota and some defines. The output is the same,it's just a simplification of the tool chain

**Reply by s-hadinger** [reply]
I just pushed a PR
